internal/ogtags/ogtags.go

```diff
@@ -71,37 +76,50 @@ func NewOGTagCache(target string, ogPassthrough bool, ogTimeToLive time.Duration
 
 	return &OGTagCache{
 		cache:               decaymap.New[string, map[string]string](),
-		targetURL:           parsedTargetURL, // Store the parsed URL
+		targetURL:           parsedTargetURL,
 		ogPassthrough:       ogPassthrough,
 		ogTimeToLive:        ogTimeToLive,
-		ogCacheConsiderHost: ogTagsConsiderHost, // todo: refactor to be a separate struct
+		ogCacheConsiderHost: ogTagsConsiderHost,
 		approvedTags:        defaultApprovedTags,
 		approvedPrefixes:    defaultApprovedPrefixes,
 		client:              client,
+		unixPrefix:          "http://unix",
 	}
 }
 
 // getTarget constructs the target URL string for fetching OG tags.
-// For Unix sockets, it creates a "fake" HTTP URL that the custom dialer understands.
+// Optimized to minimize allocations by building strings directly.
 func (c *OGTagCache) getTarget(u *url.URL) string {
+	var escapedPath = u.EscapedPath() // will cause an allocation if path contains special characters
 	if c.targetURL.Scheme == "unix" {
-		// The custom dialer ignores the host, but we need a valid http URL structure.
-		// Use "unix" as a placeholder host. Path and Query from original request are appended.
-		fakeURL := &url.URL{
-			Scheme:   "http", // Scheme must be http/https for client.Get
-			Host:     "unix", // Arbitrary host, ignored by custom dialer
-			Path:     u.Path,
-			RawQuery: u.RawQuery,
+		// Build URL string directly without creating intermediate URL object
+		var sb strings.Builder
+		sb.Grow(len(c.unixPrefix) + len(escapedPath) + len(u.RawQuery) + querySeparatorLength) // Pre-allocate
+		sb.WriteString(c.unixPrefix)
+		sb.WriteString(escapedPath)
+		if u.RawQuery != "" {
+			sb.WriteByte('?')
+			sb.WriteString(u.RawQuery)
 		}
-		return fakeURL.String()
+		return sb.String()
 	}
 
-	// For regular http/https targets
-	target := *c.targetURL // Make a copy
-	target.Path = u.Path
-	target.RawQuery = u.RawQuery
-	return target.String()
+	// For regular http/https targets, build URL string directly
+	var sb strings.Builder
+	// Pre-calculate size: scheme + "://" + host + path + "?" + query
+	estimatedSize := len(c.targetURL.Scheme) + schemeSeparatorLength + len(c.targetURL.Host) + len(escapedPath) + len(u.RawQuery) + querySeparatorLength
```

Comment:
Doesn't `net/url` [do the same thing?](https://cs.opensource.google/go/go/+/refs/tags/go1.24.4:src/net/url/url.go;drc=bb5339196f86057ffa6042d31f52a44bf7f13752;l=846)